src/App.js

```diff
@@ -1,16 +1,38 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likes, setTotalLikes] = useState(0);
+
+  const changeLikeButton = (id) => {
+    for (const message of messages) {
+      if (message.id === id) {
+        message.liked = !message.liked;
+        if (message.liked === true) {
+          setTotalLikes(likes + 1);
+        } else {
+          setTotalLikes(likes - 1);
+        }
+      }
+      const updatedMessages = [...messages];
+      setMessages(updatedMessages);
+    }
```

Comment:
This code is making changes to the state var and then creating a shallow copy and then updating the state variable from the shallow copy.  It's fine to use a shallow copy, but it should be done before making changes.  Changes can then be made to the copy of the state variable.  Despite the fact that the changes are being made to the same objects in memory, in React it's recommended not to make changes to the state variables.

```suggestion
    const updatedMessages = [...messages];
    for (const message of updatedMessages) {
      if (message.id === id) {
        message.liked = !message.liked;
        if (message.liked === true) {
          setTotalLikes(likes + 1);
        } else {
          setTotalLikes(likes - 1);
        }
      }
      setMessages(updatedMessages);
    }
```